README.md

```diff
@@ -188,9 +186,207 @@ The procedure is described in `experiments/splash/plot.py`.
+
+    # load the results into a DataFrame
+    helpers.debug("Loading data")
+    df = read_csv(infile, usecols=all_columns)
+    if df.empty:
+        helpers.error_exit(1, "The input file is empty or not readable")
+
+    # aggregate the results of repeated experiments (i.e., average across all runs of the same experiment)
+    helpers.debug("Processing results")
+    df = DataFrame({data_column: df.groupby(metadata_columns)[data_column].apply(stats.gmean, axis=0)})
+    df = df.reset_index()
+
+    # cleanup
+    df.dropna(inplace=True)
+
+    # To MB
+    # TODO 1 MB = 1024 or 1000?
+    df["memory"] = df["memory"] / 1024
+
+    # restructure the table for easier plotting and calculate the overall mean values across all benchmarks
+    pivoted = df.pivot_table(
+        index="benchmark",
+        columns=["type", "subtype", "thread_count"],
+        values=data_column,
+        margins=True,
+        aggfunc=stats.gmean,
+        margins_name="mean"
+    )
+    df = DataFrame(pivoted.to_records()).set_index("benchmark", drop=True)
+
+    # rename builds
+    df.rename(columns=RENAMINGS_ABSOLUTE, inplace=True)
+
+    # the resulting table
+    helpers.debug("Plot data\n" + str(df))
+
+    # build the plot
+    helpers.debug("Building a plot")
+    plt = plot.BarPlot(style=SplashBarPlotStyle)
+    plt.build(df,
+        title="Memory usage compared with GCC -O3 optimizations",
+        ylabel="Memory usage (in MB)",
+        vline_position=11.5
+        )
+
+    plt.savefig(
+        outfile,
+        dpi="figure",
+        pad_inches=0.1,
+        bbox_inches='tight'
+    )
+### End of insert ###
+```
 
+You can test the new code with the following commands:
+```shell script
+$ fex2 run splash -b gcc_native gcc_optimized -t time -m 1 4 -r 10 -o splash-raw-2.txt
+$ fex2 collect splash -t time -i splash-raw-2.txt -o splash-collected-2.csv
+$ fex2 plot splash -p absolute -i splash-collected-2.csv -o splash-2.pdf
+```
+Which should give you the following graph:
+![Graph of example 2](readme/example2_graph.png)
+
+## Example 2.2: Benchmark a smaller run
+
+To run smaller benchmarks, the input file of an experiment has to be modified.
+This can be done by modifying the column `default` of the input table.
+The input table `input.csv` can be found in the respective experiment folder.
+In this example we are going to adapt the splash benchmark.
+The splash input table is located at `experiments/splash/input.csv`.
+
+The `test` column of the input table has easier benchmark inputs then the `default` column.
+We can therefore copy the inputs of the test column into the default column to get smaller benchmarks.
+With that, the `ocean` benchmark is for instance changed from `-p$thread_count -n258` to `-p$thread_count -n18`.
+
+You can test the new inputs with the normal run command:
+```shell script
+$ fex2 run splash -b gcc_native gcc_optimized -t time -m 1 4 -r 10 -o splash-raw-2.txt
+```
+Which should now run a bit faster.
```

Comment:
Oh, I forgot that I did not implement this feature. The user should not modify `input.csv`. Instead, the type should be selected on command line, as an argument to `fex2 run ...`. Something like `fex2 run ... --benchmark-input test`.

Let's leave this section of readme as-is for the time being, but this feature will need to be implemented eventually.